crates/ide_completion/src/completions/fn_param.rs

```diff
@@ -143,4 +159,25 @@ fn foo2($0) {}
             "#]],
         )
     }
+
+    #[test]
+    fn test_param_completion_self_param() {
+        check(
+            r#"
+                impl A {
```

Comment:
Maybe add a `struct A;` and remove the unused methods.

Reply:
So shall I change the test to only contain self suggestions then? I was testing out if both completions work together.

Reply:
I'd remove `bar`, at least.